packages/instantsearch.js/src/lib/utils/hydrateRecommendCache.ts

```diff
@@ -0,0 +1,19 @@
+import type { InitialResults } from '../../types';
+import type { AlgoliaSearchHelper } from 'algoliasearch-helper';
+
+export function hydrateRecommendCache(
+  helper: AlgoliaSearchHelper,
+  initialResults: InitialResults
+) {
+  const recommendCache = Object.keys(initialResults).reduce(
+    (acc, indexName) => {
+      const initialResult = initialResults[indexName];
+      if (initialResult.recommendResults) {
+        return Object.assign(acc, initialResult.recommendResults.results);
```

Comment:
can go back to spread then (but maybe leave a comment about `const recommendCache = Object.assign({}, ...Object.keys(initialResults).map(key => initialResults[key].recommendResults.results))` as the next major option, seems like that's quicker still (difference much bigger in safari than in chrome though)

https://jsbench.me/9zlworivz7/1

<img width="1302" alt="Screenshot 2024-05-27 at 11 27 57" src="https://github.com/algolia/instantsearch/assets/6270048/c2976cbc-1770-4a93-beb4-d118c0a1ccfa">